mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,81 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
+    return res.status(403).jssosn({
+      error: 'Access denied',
+    });
+  }
+  next();
+};
+
+exports.forgotPassword = (req, res) => {
+  if (!req.body) return res.status(400).json({ message: 'No request body' });
+  if (!req.body.email)
```

Comment:
Instead of just checking if `req.body.email` is falsy, consider explicitly checking if it's `undefined` or `null` to avoid potential issues.